lib/knife_cookbook_doc/resource_model.rb

```diff
@@ -52,14 +52,18 @@ def attribute_description(attribute)
 
     def attribute_has_default_value?(attribute)
       specification = @native_resource.attribute_specifications[attribute]
-      specification && specification.has_key?(:default)
+      specification && specification.key?(:default)
     end
 
     def attribute_default_value(attribute)
-      if attribute_has_default_value?(attribute)
-        return @native_resource.attribute_specifications[attribute][:default]
+      default = if attribute_has_default_value?(attribute)
+                  @native_resource.attribute_specifications[attribute][:default]
+                end
+
+      if default.is_a?(Chef::DelayedEvaluator)
```

Comment:
I haven't had a chance to check this morning but what version of Chef did `Chef::DelayedEvaluator` appear in? If it is early enough version we can leave the code as is but otherwise we may need an additional check such as `Chef.const_defined?('DelayedEvaluator')`

Also you should add the change to CHANGELOG and give youself some credit there ;)

Thanks

Reply:
It was introduced in 11.6.0. While people may still be running that old of a version of chef on the servers, I would hope that their development env, where they run `knife cookbook doc`, is not that ancient.